File: Backend/src/main/java/com/bachld/project/backend/dto/request/donhoandoan/DonHoanDoAnRequest.java
package com.bachld.project.backend.dto.request.donhoandoan;

import jakarta.validation.constraints.NotBlank;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.springframework.web.multipart.MultipartFile;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class DonHoanDoAnRequest {

    @NotBlank(message = "INVALID_VALIDATION")
    String lyDo;

    MultipartFile minhChungFile; // optional
}

File: Android/app/src/main/java/com/bachld/android/data/dto/request/auth/donhoandoan/DonHoanDoAnRequest.kt
package com.bachld.android.data.dto.request.donhoandoan

import android.net.Uri

/**
 * Request dùng ở Android. `minhChungUri` chỉ để repo chuyển sang MultipartBody.Part.
 */
data class DonHoanDoAnRequest(
    val lyDo: String,
    @Transient val minhChungUri: Uri? = null
)
